cmd/osm-injector/osm-injector.go

```diff
@@ -141,8 +142,10 @@ func main() {
 	}
 	log.Info().Msgf("Initial MeshConfig %s: %v", osmMeshConfigName, meshConfig)
 
+	policyClient := policyV1alpha1Client.NewForConfigOrDie(kubeConfig)
```

Comment:
osm-injector doesn't need this. Let's just pass `nil` to be explicit with a comment?

Reply:
I propose we do not merge code refactors while we are working on a dedicated feature that is behind a feature flag. We can create an issue in OSM for improvements like these and address them in the future but lets scope the work such that it becomes to deliver on what we have planned.

Additional improvements are always welcome but they can be done in the future, with the correct tracking in place

Reply:
@shashankram done, and split up the interfaces such the injector gets the minimal set.

```
// Controller implements the full set of controller methods, including those for CRD's.
type Controller interface {
	EgressController
	BasicController
}
```

@snehachhabria I'm fine with whatever, although if we can separately agree that this is a change in the right direction, then it doesn't need to be considered in concert with the multi cluster work